tests/legacy-cli/e2e/utils/process.ts

```diff
@@ -197,22 +196,31 @@ export function waitForAnyProcessOutputToMatch(
 }
 
 export async function killAllProcesses(signal = 'SIGTERM'): Promise<void> {
-  await Promise.all(
-    _processes.map(
-      ({ pid }) =>
-        new Promise<void>((resolve, reject) => {
-          treeKill(pid, signal, (err) => {
-            if (err) {
-              reject(err);
-            } else {
-              resolve();
-            }
-          });
-        }),
-    ),
-  );
+  const processesToKill: Promise<void>[] = [];
+
+  while (_processes.length) {
+    const childProc = _processes.pop();
+    if (!childProc) {
+      continue;
+    }
+
+    processesToKill.push(
+      new Promise<void>((resolve, reject) => {
+        treeKill(childProc.pid, signal, (err) => {
+          if (err && !err.message.includes('not found')) {
+            // Ignore process not found errors.
+            // This is due to a race condition with the `waitForMatch` logic.
+            // where promises are resolved on matches and not when the process terminates.
+            reject(err);
+          } else {
+            resolve();
+          }
+        });
+      }),
+    );
+  }
 
-  _processes = [];
```

Comment:
Since this was after the `await` I guess that means processes could have been added to this array during the `await` and then we'd clear it here and lose those added processes. Was that the cause of one of the issues? (in addition to the `'not found'`)